### PR TITLE
#4212 Stock indicator in Sales Orderline 

### PR DESCRIPTION
**Pull request by cristinamghita**
#4212 Stock indicator in Sales Orderline

### CODE REVIEW

File: de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/modelvalidator/C_OrderLine.java [hunk 1]
```diff
@@ -59,4 +69,28 @@ public void deleteOlas(final I_C_OrderLine ol)
 			InterfaceWrapperHelper.delete(ola);
 		}
 	}
+
+	@ModelChange(timings = {ModelValidator.TYPE_BEFORE_CHANGE, ModelValidator.TYPE_BEFORE_NEW })
+	public void updateQtyAvailableToPromise(final I_C_OrderLine ol)
+	{
+		 final AvailableToPromiseRepository stockRepository = Adempiere.getBean(AvailableToPromiseRepository.class);
+		 final AvailableToPromiseQueryBuilder stockQueryBuilder = AvailableToPromiseQuery.builder()
+					.warehouseIds(ImmutableSet.of(ol.getM_Warehouse_ID()))
```
Comment (metas-ts) [inline de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/modelvalidator/C_OrderLine.java]: C_OrderLine.M_Warehouse_ID is not mandatory..are you sure that it's always set?

Comment (cristinamghita) [reply de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/modelvalidator/C_OrderLine.java]: @metas-ts We have this callout de.metas.adempiere.callout.OrderLine.warehouse(ICalloutField).
Plus , the default value for AD_Column C_Orderline.M_Warewhouse_ID is @M_Warehouse_ID@
I think the warehouse is always set in order line.

**Comment by metas-ts** [comment]
@hi @cristinamghita please check and (dev-)test my change here https://github.com/metasfresh/metasfresh/pull/4228/commits/e681e45adbfdce9d1c31573ca12d56a3c71b5c68

i think the old version wasn't sufficient

File: de.metas.material/dispo-commons/src/main/java/de/metas/material/dispo/commons/interceptor/C_OrderLine.java [hunk 2]
```diff
@@ -29,39 +18,26 @@
...
  * #L%
  */
 @Interceptor(I_C_OrderLine.class)
-@Component("MaterialDispoOrderLineInterceptor")
+@Component("de.metas.material.dispo.commons.interceptor.C_OrderLine")
 public class C_OrderLine
 {
-	@ModelChange(timings = {ModelValidator.TYPE_BEFORE_CHANGE, ModelValidator.TYPE_BEFORE_NEW })
-	public void updateQtyAvailableToPromise(final I_C_OrderLine ol)
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE }, ifColumnsChanged = { I_C_OrderLine.COLUMNNAME_M_Product_ID })
```
Comment (metas-ts) [inline de.metas.material/dispo-commons/src/main/java/de/metas/material/dispo/commons/interceptor/C_OrderLine.java]: hi @cristinamghita, if we update on M_Product_ID, let's also update on M_AttributeSetInstance_ID, and update all order lines if C_Order.PreparationDate changes

**Comment by metas-ts** [comment]
hi @cristinamghita please make sure you have our formatting settings from https://github.com/metasfresh/metasfresh-dev/tree/master/eclipse-config (see http://docs.metasfresh.org/developers_collection/en/getting_started_ide_en.html )

**Pull request merged**